tests/modules/test_time.py

```diff
@@ -177,6 +176,131 @@ def test_struct_time_repr(self):
+        """
+
+        seq = (1970,1,1,0,0,0,0,0,0)
+
+        test_str = adjust("""
+        print('>>> import time')
+        import time
+        print('>>> seq = {seq}')
+        seq = {seq}
+        print(">>> time.mktime({seq})")
+        print(time.mktime(seq))
+        #################################
+        print(">>> seq = time.struct_time({seq})")
+        seq = time.struct_time({seq})
+        print(">>> time.mktime(seq)")
+        print(time.mktime(seq))
+        """).format(seq=str(seq))
+
+        self.assertCodeExecution(test_str)
+
+    def test_mktime_bad_input(self):
+        """
+        When the argument passed is not tuple or struct_time.
+        """
+
+        seed = (1970,1,1,0,0,0,0,0,0)
+
+        data = [
+            False,
+            1j,
+            {key: key for key in seed},
+            1.2,
+            frozenset(seed),
+            1,
+            list(seed),
+            range(1,2,3),
+            set(seed),
+            slice(1,2,3),
+            '123456789',
+            None,
+            NotImplemented
+        ]
+
+        test_str = adjust("""
+        print('>>> import time')
+        import time
+        """)
+
+        tests = [mktime_setup(str(d)) for d in data]
+        test_str += ''.join(tests)
+
+        self.assertCodeExecution(test_str)
+
+    def test_mktime_non_integer_types(self):
+        """
+        Tests behavior when non integer types are part of the sequence passed
+        """
+
+        test_str = adjust("""
+        print('>>> import time')
+        import time
+        """)
+
+        strange_types = [ SAMPLE_DATA[type][0] for type in SAMPLE_DATA if type != 'int' ]
+
+        tests = [ mktime_setup( str((1970, t, 1, 2, 0, 0, 0, 0, 0)) ) for t in strange_types]
+        test_str += ''.join(tests)
+
+        self.assertCodeExecution(test_str)
+
+    def test_mktime_seq_wrong_length(self):
+        """
+        Sequence has the wrong length
+        """
+
+        seq = (1970,1,1,0,0,0,0,0)  # length=8
+
+        test_str = adjust("""
+        print('>>> import time')
+        import time
+        """)
+
+        test_str += mktime_setup(seq)
+        print(test_str)
+        self.assertCodeExecution(test_str)
+
+    def test_mktime_wrong_arg_count(self):
+        """
+        Called with the wrong number of args
+        """
+
+        test_str = adjust("""
+        print('>>> import time')
+        import time
+        print('>>> time.mktime()')
+        print(time.mktime())
+        ############################
+        print('>>> time.mktime(1,2)')
+        print(time.mktime(1,2))
+        """)
+
+        self.assertCodeExecution(test_str)
+
+    @unittest.expectedFailure
+    def test_mktime_overflow_error(self):
+        """
+        tests OverflowError on dates earlier than an arbitrarily defined date
+        for now this is defined as 1900-01-01
+
+        currently this can't be garuneteed to pass because it doesn't not account for behavior across platforms.
```

Comment:
I'd suggest the best approach here is to use `assertJavaScriptExecution()` rather than `assertCodeExecution`. The rest of this test suite will validate that the behavior of mktime is consistent with CPython; all this test needs to do is ensure that an error will be raised under known conditions. It's worth a documentation note to describe *why* this test can't be performed in a cross platform way; but that's all you really need to do.